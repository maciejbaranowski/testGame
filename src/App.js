import React, { Component } from "react";
import PropTypes from "prop-types";
import Game from "./Game";
import {
  Button,
  ProgressBar,
  Grid,
  Row,
  Col,
  Badge,
  Alert,
  Fade,
  ListGroup,
  ListGroupItem
} from "react-bootstrap";

const PlayerStats = props => (
  <div>
    <h3>Player statistics:</h3>
    <ul>
      <li key="health">
        Health:{" "}
        <ProgressBar
          now={props.health}
          label={props.health + "/100"}
          bsStyle="danger"
        />
      </li>
      <li key="cash">
        Cash: <Badge>{props.cash}</Badge>
      </li>
    </ul>
    <Fade in={props.status.length > 0}>
      <Alert bsStyle="danger">{props.status}</Alert>
    </Fade>
  </div>
);
PlayerStats.propTypes = {
  health: PropTypes.number.isRequired,
  cash: PropTypes.number.isRequired,
  status: PropTypes.string
};

const Action = props => (
  <div>
    <Button onClick={props.callback} disabled={props.isDisabled}>
      {props.text}
    </Button>
  </div>
);

Action.propTypes = {
  callback: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired
};

const Actions = props => {
  return (
    <div>
      <h3>Actions:</h3>
      <Action
        callback={() => {
          props.updateGame(props.game.player.fight());
        }}
        text={"Fight!"}
        isDisabled={!props.game.player.canFight()}
      />
    </div>
  );
};

Actions.propTypes = {
  game: PropTypes.object.isRequired,
  updateGame: PropTypes.func.isRequired
};

const Log = props => (
  <div>
    <h3>Game log:</h3>
    <ListGroup>
      {props.log.map(logText => {
        return <ListGroupItem>{logText}</ListGroupItem>;
      })}
    </ListGroup>
  </div>
);

class App extends Component {
  constructor() {
    super();
    this.state = {
      game: new Game()
    };
  }
  render() {
    return (
      <Grid className="App">
        <Row>
          <Col md={4}>
            <PlayerStats
              health={this.state.game.player.health}
              cash={this.state.game.player.cash}
              status={this.state.game.player.status}
            />
          </Col>
          <Col md={4}>
            <Actions
              game={this.state.game}
              updateGame={newState => {
                this.setState({ game: newState });
              }}
            />
          </Col>
          <Col md={4}>
            <Log log={this.state.game.log} />
          </Col>
        </Row>
      </Grid>
    );
  }
}

export default App;
