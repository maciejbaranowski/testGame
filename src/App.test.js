import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders as before', () => {
  let output = renderer.create(<App/>).toJSON();
  expect(output).toMatchSnapshot();
});
