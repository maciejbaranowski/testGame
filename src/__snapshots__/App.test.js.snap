// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as before 1`] = `
<div
  className="App container"
>
  <div
    className="row"
  >
    <div
      className="col-md-4"
    >
      <div>
        <h3>
          Player statistics:
        </h3>
        <ul>
          <li>
            Health:
             
            <div
              className="progress"
            >
              <div
                aria-valuemax={100}
                aria-valuemin={0}
                aria-valuenow={100}
                className="progress-bar progress-bar-danger"
                role="progressbar"
                style={
                  Object {
                    "width": "100%",
                  }
                }
              >
                100/100
              </div>
            </div>
          </li>
          <li>
            Cash: 
            <span
              className="badge"
            >
              10
            </span>
          </li>
        </ul>
        <div
          className="fade alert alert-danger"
          role="alert"
        >
          
        </div>
      </div>
    </div>
    <div
      className="col-md-4"
    >
      <div>
        <h3>
          Actions:
        </h3>
        <div>
          <button
            className="btn btn-default"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Fight!
          </button>
        </div>
      </div>
    </div>
    <div
      className="col-md-4"
    >
      <div>
        <h3>
          Game log:
        </h3>
        <ul
          className="list-group"
        />
      </div>
    </div>
  </div>
</div>
`;
